app/helpers/fhir_helper.rb

```diff
@@ -432,6 +456,38 @@ def laboratory_from_fhir(observation)
     }
   end
 
+  # Returns a representative FHIR::QuestionnaireResponse for an instance of a Sara Alert Assessment.
+  # https://www.hl7.org/fhir/questionnaireresponse.html
+  def assessment_as_fhir(assessment)
+    FHIR::QuestionnaireResponse.new(
+      meta: FHIR::Meta.new(lastUpdated: assessment.updated_at.strftime('%FT%T%:z')),
+      id: assessment.id,
+      subject: FHIR::Reference.new(reference: "Patient/#{assessment.patient_id}"),
+      status: 'completed',
+      extension: [
+        to_bool_extension(assessment.symptomatic, 'symptomatic'),
+        to_datetime_extension(assessment.created_at, 'created-at'),
+        to_string_extension(assessment.who_reported, 'who-reported')
+      ],
+      item: assessment.reported_condition.symptoms.enum_for(:each_with_index).collect do |s, index|
+        case s.type
+        when 'IntegerSymptom'
+          FHIR::QuestionnaireResponse::Item.new(text: s.name,
+                                                answer: FHIR::QuestionnaireResponse::Item::Answer.new(valueInteger: s.int_value),
+                                                linkId: index.to_s)
+        when 'FloatSymptom'
+          FHIR::QuestionnaireResponse::Item.new(text: s.name,
+                                                answer: FHIR::QuestionnaireResponse::Item::Answer.new(valueDecimal: s.float_value),
+                                                linkId: index.to_s)
+        when 'BoolSymptom'
+          FHIR::QuestionnaireResponse::Item.new(text: s.name,
+                                                answer: FHIR::QuestionnaireResponse::Item::Answer.new(valueBoolean: s.bool_value),
+                                                linkId: index.to_s)
```

Comment:
I realized this bit was just moved, but I think this is important enough to include a change here:

For each of these, in the case where the symptom is `nil` (in practice, when the assessment was responded to over a medium doesn't allow for indication of specific symptoms), as it stands the `answer` will just be omitted. I think it would be preferable to more explicitly state that the symptom answer is unknown, which could be done by using a `valueCoding` and using the [`NullFlavor`](http://hl7.org/fhir/v3/NullFlavor/cs.html) `UNK` code.

Reply:
Good catch, fixed in https://github.com/SaraAlert/SaraAlert/pull/1016/commits/3881a715e296d7607ca4ae30ad7a13805b9c569c.